<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>MainWindow</class>
    <widget class="QMainWindow" name="MainWindow">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>640</width>
                <height>780</height>
            </rect>
        </property>
        <property name="minimumSize">
            <size>
                <width>500</width>
                <height>650</height>
            </size>
        </property>
        <property name="maximumSize">
            <size>
                <width>1200</width>
                <height>1400</height>
            </size>
        </property>
        <property name="windowTitle">
            <string>Sudoku Solver ‚Ä¢ Qt6</string>
        </property>
        <property name="styleSheet">
            <string notr="true">/* ‚îÄ‚îÄ Genel ‚îÄ‚îÄ */
                QMainWindow { background-color: #1A1A2E; }
                QWidget { background-color: #1A1A2E; }

                /* ‚îÄ‚îÄ Ba≈ülƒ±k ‚îÄ‚îÄ */
                QLabel#titleLabel {
                color: #E94560;
                font-size: 28px;
                font-weight: bold;
                letter-spacing: 1px;
                padding: 4px;
                }

                /* ‚îÄ‚îÄ S√ºre etiketi ‚îÄ‚îÄ */
                QLabel#timeLabel {
                color: #80DEEA;
                font-size: 15px;
                }

                /* ‚îÄ‚îÄ Durum mesajƒ± ‚îÄ‚îÄ */
                QLabel#statusLabel {
                color: #B0BEC5;
                font-size: 13px;
                padding: 4px;
                }

                /* ‚îÄ‚îÄ Buton temel stili ‚îÄ‚îÄ */
                QPushButton {
                border: 2px solid transparent;
                border-radius: 10px;
                font-size: 14px;
                font-weight: bold;
                padding: 10px 16px;
                color: #FFFFFF;
                }
                QPushButton:pressed {
                border: 2px inset rgba(0, 0, 0, 0.3);
                }
                QPushButton:disabled {
                color: #90CAF9;
                opacity: 0.6;
                }

                /* ‚îÄ‚îÄ √á√∂z Butonu ‚îÄ‚îÄ */
                QPushButton#solveBtn {
                background-color: #1565C0;
                border-color: #1976D2;
                }
                QPushButton#solveBtn:hover {
                background-color: #1E88E5;
                border-color: #2196F3;
                border: 2px solid #2196F3;
                }
                QPushButton#solveBtn:pressed {
                background-color: #1565C0;
                border: 2px inset rgba(0, 0, 0, 0.3);
                }

                /* ‚îÄ‚îÄ Animasyon Butonu ‚îÄ‚îÄ */
                QPushButton#animateBtn {
                background-color: #6A1B9A;
                border-color: #7B1FA2;
                }
                QPushButton#animateBtn:hover {
                background-color: #7B1FA2;
                border-color: #8E24AA;
                border: 2px solid #8E24AA;
                }
                QPushButton#animateBtn:pressed {
                background-color: #6A1B9A;
                border: 2px inset rgba(0, 0, 0, 0.3);
                }

                /* ‚îÄ‚îÄ Yeni Puzzle Butonu ‚îÄ‚îÄ */
                QPushButton#newPuzzleBtn {
                background-color: #00695C;
                border-color: #00796B;
                }
                QPushButton#newPuzzleBtn:hover {
                background-color: #00796B;
                border-color: #00897B;
                border: 2px solid #00897B;
                }
                QPushButton#newPuzzleBtn:pressed {
                background-color: #00695C;
                border: 2px inset rgba(0, 0, 0, 0.3);
                }

                /* ‚îÄ‚îÄ Temizle Butonu ‚îÄ‚îÄ */
                QPushButton#clearBtn {
                background-color: #B71C1C;
                border-color: #C62828;
                }
                QPushButton#clearBtn:hover {
                background-color: #C62828;
                border-color: #D32F2F;
                border: 2px solid #D32F2F;
                }
                QPushButton#clearBtn:pressed {
                background-color: #B71C1C;
                border: 2px inset rgba(0, 0, 0, 0.3);
                }
            </string>
        </property>
        <widget class="QWidget" name="centralwidget">
            <layout class="QVBoxLayout" name="mainLayout">
                <property name="leftMargin">
                    <number>24</number>
                </property>
                <property name="topMargin">
                    <number>20</number>
                </property>
                <property name="rightMargin">
                    <number>24</number>
                </property>
                <property name="bottomMargin">
                    <number>20</number>
                </property>
                <property name="spacing">
                    <number>14</number>
                </property>
                <!-- Ba≈ülƒ±k -->
                <item>
                    <widget class="QLabel" name="titleLabel">
                        <property name="text">
                            <string>üß© Sudoku Solver</string>
                        </property>
                        <property name="alignment">
                            <set>Qt::AlignCenter</set>
                        </property>
                    </widget>
                </item>
                <!-- Sudoku Grid (Custom Widget) -->
                <item alignment="Qt::AlignCenter">
                    <widget class="SudokuGrid" name="grid">
                        <property name="sizePolicy">
                            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                                <horstretch>0</horstretch>
                                <verstretch>0</verstretch>
                            </sizepolicy>
                        </property>
                        <property name="minimumSize">
                            <size>
                                <width>506</width>
                                <height>506</height>
                            </size>
                        </property>
                        <property name="maximumSize">
                            <size>
                                <width>506</width>
                                <height>506</height>
                            </size>
                        </property>
                    </widget>
                </item>
                <!-- Zaman Etiketi -->
                <item>
                    <widget class="QLabel" name="timeLabel">
                        <property name="text">
                            <string>‚è± S√ºre: ‚Äî</string>
                        </property>
                        <property name="alignment">
                            <set>Qt::AlignCenter</set>
                        </property>
                    </widget>
                </item>
                <!-- Status Etiketi -->
                <item>
                    <widget class="QLabel" name="statusLabel">
                        <property name="text">
                            <string>Hazƒ±r</string>
                        </property>
                        <property name="maximumHeight">
                            <number>40</number>
                        </property>
                        <property name="wordWrap">
                            <bool>true</bool>
                        </property>
                        <property name="alignment">
                            <set>Qt::AlignCenter</set>
                        </property>
                    </widget>
                </item>
                <!-- Butonlar -->
                <item>
                    <layout class="QHBoxLayout" name="buttonLayout">
                        <property name="spacing">
                            <number>10</number>
                        </property>
                        <item>
                            <widget class="QPushButton" name="solveBtn">
                                <property name="minimumHeight">
                                    <number>46</number>
                                </property>
                                <property name="text">
                                    <string>‚ö° √á√∂z</string>
                                </property>
                                <property name="cursor">
                                    <cursorShape>PointingHandCursor</cursorShape>
                                </property>
                            </widget>
                        </item>
                        <item>
                            <widget class="QPushButton" name="animateBtn">
                                <property name="minimumHeight">
                                    <number>46</number>
                                </property>
                                <property name="text">
                                    <string>üé¨ Animasyon</string>
                                </property>
                                <property name="cursor">
                                    <cursorShape>PointingHandCursor</cursorShape>
                                </property>
                            </widget>
                        </item>
                        <item>
                            <widget class="QPushButton" name="newPuzzleBtn">
                                <property name="minimumHeight">
                                    <number>46</number>
                                </property>
                                <property name="text">
                                    <string>üé≤ Yeni Puzzle</string>
                                </property>
                                <property name="cursor">
                                    <cursorShape>PointingHandCursor</cursorShape>
                                </property>
                            </widget>
                        </item>
                        <item>
                            <widget class="QPushButton" name="clearBtn">
                                <property name="minimumHeight">
                                    <number>46</number>
                                </property>
                                <property name="text">
                                    <string>üóë Temizle</string>
                                </property>
                                <property name="cursor">
                                    <cursorShape>PointingHandCursor</cursorShape>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </item>
            </layout>
        </widget>
    </widget>
    <customwidgets>
        <customwidget>
            <class>SudokuGrid</class>
            <extends>QWidget</extends>
            <header>SudokuGrid.hpp</header>
        </customwidget>
    </customwidgets>
</ui>

